### PR TITLE
Bug fix for db schema and model

### PR DESCRIPTION
**Pull request by wl2484**
Fix bugs found in models.py

### CODE REVIEW

File: service/models.py [hunk 1]
```diff
@@ -1,227 +1,219 @@
...
-                "Invalid Item: missing " + error.args[0]
+                f"Invalid {type(self).__name__}: missing {error.args[0]}"
             ) from error
         except TypeError as error:
             raise DataValidationError(
-                "Invalid Item: body of request contained bad or no data - "
-                "Error message: " + error
+                f"Invalid {type(self).__name__}: failed to deserialize request body\nError message: {error}"
             ) from error
         return self
```
Comment (wl2484) [inline service/models.py:203]: heads-up 🙋‍♀️ `return self` can be eliminated

File: service/models.py [hunk 2]
```diff
@@ -1,227 +1,219 @@
...
-                "Invalid Shopcart: missing " + error.args[0]
+                f"Invalid {type(self).__name__}: missing {error.args[0]}"
             ) from error
         except TypeError as error:
             raise DataValidationError(
-                "Invalid Shopcart: body of request contained bad or no data - "
-                "Error message: " + error
+                f"Invalid {type(self).__name__}: failed to deserialize request body\nError message: {error}"
             ) from error
         return self
```
Comment (wl2484) [inline service/models.py:104]: heads-up 🙋‍♀️ `return self` can be eliminated

File: service/models.py [hunk 3]
```diff
@@ -1,227 +1,219 @@
...
-            "customer_name": self.customer_name,
+        """ Serialize a shopcart object into a dictionary """
+        shopcart = {
+            "id": self.id,
+            "name": self.name,
+            "items": list()
         }
+        for item in self.items:
+            shopcart["items"].append(item.serialize())
+        return shopcart
```
Comment (LillisLuo) [inline service/models.py:79]: Almost the same in the serialize function but I wrote **"items": []** at line 75

Comment (wl2484) [reply service/models.py:79]: ohhh, I'm good with both! Feel free to change it to `"items": []` 🙌🏻

File: service/models.py [hunk 4]
```diff
@@ -1,227 +1,219 @@
...
-
-    @classmethod
-    def find_by_name(cls, name):
-        """Returns all Items with the given name
-
-        Args:
-            name (string): the name of the Item you want to match
-        """
-        logger.info("Processing name query for %s ...", name)
-        return cls.query.filter(cls.name == name)
```
Comment (LillisLuo) [inline service/models.py:227]: What about comment the two functions for now since we may use them later?

**Review by LillisLuo — APPROVED** [review]

**Comment by LillisLuo** [comment]
Thanks for your modification. All good in the two files. Approve done!

File: service/models.py (hunk 4 above)
Comment (wl2484) [reply service/models.py:227]: sounds good to me!

File: service/models.py [hunk 5]
```diff
@@ -1,227 +1,227 @@
...
-        logger.info("Processing name query for %s ...", name)
-        return cls.query.filter(cls.customer_name == name) 
-    
+        logger.info(f"Get {cls.__name__} by name={name}")
+        return cls.query.filter(cls.name == name)
+
+    @classmethod
+    def find_id(cls):
+
+        result = cls.query.with_entities(Shopcart.customer_id).all()
```
Comment (wl2484) [inline service/models.py]: should be `Shopcart.id` instead of `Shopcart.customer_id` now lol (sorry for causing the trouble)

Comment (LillisLuo) [reply service/models.py]: Yes, it should be **Shopcart.id** now

**Review by momo100110 — APPROVED** [review]
Looks good to me. Thank you for all of your work!

**Pull request merged**